lib/archethic.ex

```diff
@@ -80,25 +79,50 @@ defmodule Archethic do
     end
   end
 
-  defp forward_transaction(
-         tx,
-         welcome_node_key,
-         nodes \\ P2P.authorized_and_available_nodes()
-         |> Enum.filter(&Node.locally_available?/1)
-         |> P2P.nearest_nodes()
-       )
+  @spec forward_transaction(tx :: Transaction.t(), welcome_node_key :: Crypto.key()) :: :ok
+  defp forward_transaction(tx, welcome_node_key) do
+    %Node{network_patch: welcome_node_patch} = P2P.get_node_info!(welcome_node_key)
+
+    nodes =
+      P2P.authorized_and_available_nodes()
+      |> Enum.reject(&(&1.first_public_key == welcome_node_key))
+      |> Enum.sort_by(& &1.first_public_key)
+      |> P2P.nearest_nodes(welcome_node_patch)
+      |> Enum.filter(&Node.locally_available?/1)
+
+    nodes =
+      if Crypto.first_node_public_key() != welcome_node_key do
+        #  if this node is not the welcome node then select
+        # next node from the this node position in nodes list
+        current_node = Crypto.first_node_public_key()
+
+        index = Enum.find_index(nodes, &(&1.first_public_key == current_node))
+        {_l, r} = Enum.split(nodes, index + 1)
+        r
+      else
+        nodes
+      end
+
+    TaskSupervisor
+    |> Task.Supervisor.async_nolink(fn ->
```

Comment:
Why do you need other process to send the transaction? 
And also, what is the purpose of the `async`? You are not awaiting the task result.

Reply:
spawn?

Reply:
node 1 waits for node 2 to return ok , node 2 waits for node 3 to send txn and fget back response,